### PR TITLE
[FIX] fix column count for search engine scores in mzTab

### PR DESCRIPTION
**Pull request by timosachsenberg**

### CODE REVIEW

File: src/topp/MzTabExporter.cpp [hunk 1]
```diff
@@ -782,8 +782,7 @@ namespace OpenMS
         db_version = sp.db_version.empty() ? MzTabString() : MzTabString(sp.db_version);
       }
 
-      // currently we don't save how many channels we saved in a consensus feature so we scan the file
-      // for the maximum map index in a consensus feature and take this as number of study variables
+      // determine number of channels
       Size n_study_variables = consensus_map.getFileDescriptions().size();
```
Comment (lars20070) [inline src/topp/MzTabExporter.cpp]: This works for label-free, but not for labelled datasets. In the labelled case, you can have one file but three channels.

Comment (timosachsenberg) [inline src/topp/MzTabExporter.cpp]: Yes. In the label-free case you will have several files. Where is the problem?

Comment (lars20070) [inline src/topp/MzTabExporter.cpp]: In the labelled case, you can have one file but three study variables.

Comment (hendrikweisser) [inline src/topp/MzTabExporter.cpp]: The labels are also stored in the file descriptions, so there should be no problem. See http://ftp.mi.fu-berlin.de/pub/OpenMS/documentation/html/structOpenMS_1_1ConsensusMap_1_1FileDescription.html.

Comment (lars20070) [inline src/topp/MzTabExporter.cpp]: Ok. Thanks @hendrikweisser.

+1

**Comment by lars20070** [comment]
Can you please add a test using parts of the `consensusXML` @VOOginga provided?

**Comment by timosachsenberg** [comment]
I will add tests when it stabilizes - right now there are just too many things missing and a test might suggest that the mzTab generated is correct

**Comment by lars20070** [comment]
Ok.

+1

**Comment by VOOginga** [comment]
Hallo Timo
Thanks for updating the code on the MZTab Exporter but we still have a problem. The columns still do not match and some of them lack headers. Kindly have a look at this

https://gist.github.com/VOOginga/632a9702389fa22bd405

**Pull request merged**